File: src/test/java/com/chimbori/crux/TestHelper.java
package com.chimbori.crux;

import com.chimbori.crux.articles.Article;
import com.chimbori.crux.articles.ArticleExtractor;
import com.chimbori.crux.common.CharsetConverter;
import com.chimbori.crux.common.Log;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;

import static org.junit.Assert.fail;

public class TestHelper {
  private TestHelper() {
  }

  public static Article extractFromTestFile(String baseUri, String testFile) {
    try {
      Article article = ArticleExtractor.with(baseUri,
              CharsetConverter.readStream(new FileInputStream(new File("test_data/" + testFile))).content)
          .extractMetadata()
          .extractContent()
          .estimateReadingTime()
          .article();

      Log.i("%s", article.document.childNodes().toString());
      return article;
    } catch (FileNotFoundException e) {
      fail(e.getMessage());
      return null;
    }
  }
}

File: src/test/kotlin/com/chimbori/crux/TestHelper.kt
package com.chimbori.crux

import com.chimbori.crux.articles.Article
import com.chimbori.crux.articles.ArticleExtractor
import java.io.File
import java.io.FileNotFoundException
import java.nio.charset.Charset
import okhttp3.HttpUrl
import okhttp3.HttpUrl.Companion.toHttpUrl
import org.junit.Assert.fail

fun fromFile(baseUrl: String, testFile: String, charset: String? = "UTF-8") =
  fromFile(baseUrl.toHttpUrl(), testFile, charset)

fun fromFile(baseUrl: HttpUrl, testFile: String, charset: String? = "UTF-8"): Article = try {
  ArticleExtractor(baseUrl, File("test_data/$testFile").readText(Charset.forName(charset)))
    .extractMetadata()
    .extractContent()
    .estimateReadingTime()
    .article
} catch (e: FileNotFoundException) {
  fail(e.message)
  throw e
}
